Propose idea page

sensaur opened the pull request:
Add add routing to desktop/mobile navbar on tabs click &
add desktop version for propose idea page with stepper and collecting data from user.

sensaur committed:
feat: add routing to desktop navbar on tabs click

sensaur committed:
feat: add routing to mobile navbar on tabs click

sensaur committed:
feat: add propose-idea page

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+};
+
+const buttonStyles = {
+  Далее: commonContainedStyle,
```
Давай избегать русских слов в ключах массивов :) Переделай пожалуйста на их англоязычные аналоги

sensaur replied:
готово, поправил

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+          position: 'relative',
+        }}>
+        <Typography variant="h5" gutterBottom>
+          Давай подытожим 🙂
```
Сорри, забыл про это написать в задаче, давай для текстов использовать i18n чтобы мы имели возможность быстро перевести весь сервис на другой язык

sensaur replied:
Ага, пока сделал внутри компонента переключатель между русским и английским, надо подумать как оптимальнее настроить на все приложение, достаточно того что идет из коробки некст или допом i18next

tomfordrumm replied:
Не, давай сразу next-i18n прикручивать :)

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+  };
+
+  const buttonActions = {
+    Далее: handleNextClick,
```
Здесь тоже давай русский заменим на англоязычные аналоги

sensaur replied:
готово

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+          <Box sx={{ width: '100%', marginBottom: '8px' }}>
+            <Button
+              {...buttonStyles['Изменить данные']}
+              onClick={buttonActions['Изменить данные']}
```
Вот тут поправить после изменения ключей

sensaur replied:
готово

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+          Ты хочешь получить:
+        </Typography>
+        <Typography variant="subtitle1">
+          {userInputs['Опишите, что нужно сделать']}
```
Вот тут поправить после изменения полей

sensaur replied:
готово

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+              label={currentStep.labelTitle}
+              margin="normal"
+              value={userInputs[currentStep.labelTitle] || ''}
+              onChange={(e) => handleInputChange(currentStep.labelTitle, e.target.value)}
```
Вот тут вместо step Label давай передавать field name (который нужно будет добавит в stepData.json) 

Маппинг такой: 

title -> название 
description -> Что нужно сделать?
additional -> Кто ваш клиент

sensaur replied:
готово

tomfordrumm commented on src/app/propose-idea/page.tsx:
```diff
@@ -0,0 +1,395 @@
+            >
+              {currentStep.title}
+            </Typography>
+            <TextField
```
Возможно ли для description и additional использовать вот этот компонент 
https://mui.com/base-ui/react-textarea-autosize/ ?

sensaur replied:
я попробовал Textarea Autosize но поехали стили и быстро не смог сделать визуальное оформление похожее на TextField. Если нужна многострочность в инпуте то у TextField есть свойство multiline которое это позволяет, посмотри пж в текущей версии покрывает ли это то что нужно, если нет - вернемся к Textarea Autosize

tomfordrumm reviewed (changes requested):
В целом все хорошо, но есть несколько моментов, которые я описал в комментах.

sensaur committed:
fix: change keys from in objects from cyrillic to latin

sensaur committed:
fix: add fieldName to stepData and usage in page

sensaur committed:
feat: expand data structure in stepData, replace buttons from array of strings to array of objects

sensaur committed:
fix: add titles to step 4 to stepData

sensaur committed:
fix: add titles to step 4 to stepData

sensaur committed:
fix: add english language support to propose idea page

sensaur committed:
feat: add multiline to input field at steps 1 - description and step 2 additional

sensaur committed:
feat:
- move pages to [locale] folder
- add next-intl package: configure i18nMiddleware, i18n
- add LocaleContext and LocaleProvider to root component
- add my-task page using internalization const t = useTranslations('my-tasks')

sensaur committed:
feat:
- move pages to [locale] folder
- add next-intl package: configure i18nMiddleware, i18n
- add LocaleContext and LocaleProvider to root component
- add my-task page using internalization const t = useTranslations('my-tasks')

sensaur committed:
feat:
- move pages to [locale] folder
- add next-intl package: configure i18nMiddleware, i18n
- add LocaleContext and LocaleProvider to root component
- add my-task page using internalization const t = useTranslations('my-tasks')

sensaur commented:
Подрубил пакет next-intl + к нему конфиги i18nMiddleware, i18n
Перенес страницы вложив в папку [locale] 
Обернул корневой компонент в LocaleProvider чтобы класть локаль в контекст и юзать ее при рендеринге страниц на клиенте
Сделал мультиязычной страницу my-task - в зависимости от локали вывод контента через t
http://localhost:3000/ru/my-tasks
http://localhost:3000/en/my-tasks
